### PR TITLE
Add support for audit logging package operations via plugin

### PR DESCRIPTION
**Pull request by pmatilai**
This adds a plugin for creating Linux audit events for package operations (install, update, erase), along with their result code and signature check information. Initial versions in RHEL were built-in, but as this adds the necessary infrastructure + API to retrieve verification status outside rpmtsRun() guts, there's no need to drag another library dependency to the core.

### CODE REVIEW

File: plugins/audit.c [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    int enforce = (rpmtsVfyLevel(ts) & RPMSIG_SIGNATURE_TYPE) != 0;
+
+    getAuditOps(ts, ops, nelem);
+
+    for (int i = 0; i < nelem; i++) {
+	const char *op = ops[i].op;
+	if (op) {
+	    rpmte p = ops[i].te;
+	    char *nevra = audit_encode_nv_string("sw", rpmteNEVRA(p), 0);
+	    char eventTxt[256];
```
Comment (ignatenkobrain) [inline plugins/audit.c]: is there any reason to hardcode 256 bytes?

Comment (pmatilai) [reply plugins/audit.c]: That was inherited from the sample code I got from the audit folks for this. I could imagine arbitrarily long strings are not that welcome in the audit system, but I don't know if there's an actual reason for it.

Comment (pmatilai) [reply plugins/audit.c]: Anyway... these kind of hardcoded buffersizes for strings belong to the nineties, and it's actually even buggy in the case the string happened to be larger than the hardcoded buffer as the string wont be terminated in that case. Pushed an updated version with dynamically allocated string.

**Comment by pmatilai** [comment]
Figured the RPMSIG_OTHER_TYPE thing just too ugly to live and replaced it with a separate RPMSIG_UNVERIFIED_TYPE value. It's still a bit of a hack but I wont lose sleep over this.

**Comment by ffesti** [comment]
Patch itself looks good. I wonder if we want to add log entries for transaction start and end, though. But that should not stop this patch set from going in.

**Comment by ffesti** [comment]
The plugin should probably be build in the CI, too.
Adding the necessary devel packages and --with-audit to ci/Dockerfile should do the trick.

**Comment by pmatilai** [comment]
Oh, good point about the CI. We can't actually test the functionality but we can test it compiles.

**Comment by pmatilai** [comment]
Updated to include CI + the typo fix.

**Pull request merged**